content/documentation/pages/4-batch-developer-guides/2-batch/2-spring-batch.md

```diff
@@ -8,34 +8,310 @@ description: 'Create a Spring Batch Job'
+          return new BillProcessor();
+      }
+    }
+    ```
+
+    Before moving on let's look at our configuration a little bit.
+    The `@EnableBatchProcessing` annotation enables Spring Batch features and provide a base configuration for setting up batch jobs.
+    The `@EnableTask` annotation sets up a TaskRepository which stores information about the task execution such as the start and end time of the task and the exit code.
+    In the configuration above we see that our `ItemReader` bean is an instance of `JsonItemReader`. The `JsonItemReader` will read the contents of a resource and unmarshall the Json data into Usage objects. The `JsonItemReader` is one of the `ItemReader`s provided by Spring Batch.
+    We also see that our `ItemWriter` bean is an instance of `JdbcBatchItemWriter`. The `JdbcBatchItemWriter` will write the results to our database. The `JdbcBatchItemWriter` is one of the `ItemWriter`s provided by Spring Batch.
+    And the `ItemProcessor` is our very own `BillProcessor`. To make life easier notice that all the beans that use Spring Batch provided classes (`Job`, `Step`, `ItemReader`, `ItemWriter`) are being built using builders provided by Spring Batch.
 
 ### Testing
 
-Show a simple test.
+Now that we have written our code, its time to write our test. In this case we want to make sure that the bill information has been properly inserted into the `BILLING_STATEMENTS` table.
+Let’s create our test. Update the [BillrunApplicationTests.java](https://github.com/spring-cloud/spring-cloud-dataflow-samples/blob/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/test/java/io/spring/billrun/BillrunApplicationTests.java) such that looks like the contents below.
+
+```java
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@SpringBatchTest
+public class BillrunApplicationTests {
+
+	@Autowired
+	private DataSource dataSource;
+
+	private JdbcTemplate jdbcTemplate;
+
+	@Before
+	public void setup()  {
+		this.jdbcTemplate = new JdbcTemplate(this.dataSource);
+	}
+
+	@Test
+	public void testJobResults() {
+		testResult();
+	}
+
+	private void testResult() {
+		List<Bill> billStatements = this.jdbcTemplate.query("select id, " +
+						"first_name, last_name, minutes, data_usage, bill_amount " +
+						"FROM bill_statements ORDER BY id",
+				(rs, rowNum) -> new Bill(rs.getLong("id"),
+						rs.getString("FIRST_NAME"), rs.getString("LAST_NAME"),
+						rs.getLong("DATA_USAGE"), rs.getLong("MINUTES"),
+						rs.getDouble("bill_amount")));
+
+		assertEquals(5, billStatements.size());
+		Bill billStatement = billStatements.get(0);
+		assertEquals(6, billStatement.getBillAmount(), 1e-15);
+		assertEquals("jane", billStatement.getFirstName());
+		assertEquals("doe", billStatement.getLastName());
+		assertEquals(new Long(1), billStatement.getId());
+		assertEquals(new Long(500), billStatement.getMinutes());
+		assertEquals(new Long(1000), billStatement.getDataUsage());
```

Comment:
Annoying bit that my IDE does not automatically handle the static imports.  We may want to consider just importing the class for easy copy/pasting.

Reply:
Perhaps with a note to address static imports. Developers are going to notice if you do or if you don't, so we should explain why we did it. Making it work with a simpler paste is the good choice.